optimizely/odp/odp_event_manager.py

```diff
@@ -0,0 +1,233 @@
+        self.thread_exception = False
+        """thread_exception will be True if the processing thread did not exit cleanly"""
+
+    @property
+    def is_running(self) -> bool:
+        """Property to check if consumer thread is alive or not."""
+        return self.thread.is_alive()
+
+    def start(self) -> None:
+        """Starts the batch processing thread to batch events."""
+        if self.is_running:
+            self.logger.warning('ODP event queue already started.')
+            return
+
+        self.thread.start()
+
+    def _run(self) -> None:
+        """Processes the event queue from a child thread. Events are batched until
+        the batch size is met or until the flush timeout has elapsed.
+        """
+        try:
+            while True:
+                timeout = self._get_time_till_flush()
+
+                try:
+                    item = self.event_queue.get(True, timeout)
+                except Empty:
+                    item = None
+
+                if item == Signal.SHUTDOWN:
+                    self.logger.debug('Received ODP event shutdown signal.')
+                    break
+
+                elif item == Signal.FLUSH:
+                    self.logger.debug('Received ODP event flush signal.')
+                    self._flush_batch()
+                    self.event_queue.task_done()
+                    continue
+
+                elif isinstance(item, OdpEvent):
+                    self._add_to_batch(item)
+                    self.event_queue.task_done()
+
+                elif len(self._current_batch) > 0:
+                    self.logger.debug('Flushing on interval.')
+                    self._flush_batch()
+
+                else:
+                    self._set_flush_deadline()
+
+        except Exception as exception:
+            self.thread_exception = True
+            self.logger.error(f'Uncaught exception processing ODP events. Error: {exception}')
+
+        finally:
+            self.logger.info('Exiting ODP event processing loop. Attempting to flush pending events.')
+            self._flush_batch()
+            if item == Signal.SHUTDOWN:
+                self.event_queue.task_done()
+
+    def flush(self) -> None:
+        """Adds flush signal to event_queue."""
+        try:
+            self.event_queue.put_nowait(Signal.FLUSH)
+        except Full:
+            self.logger.error("Error flushing ODP event queue")
+
+    def _flush_batch(self) -> None:
+        """Flushes current batch by dispatching event.
+        Should only be called by the processing thread."""
+        self._set_flush_deadline()
+
+        batch_len = len(self._current_batch)
+        if batch_len == 0:
+            self.logger.debug('Nothing to flush.')
+            return
+
+        api_key = self.odp_config.get_api_key()
+        api_host = self.odp_config.get_api_host()
+
+        if not api_key or not api_host:
+            self.logger.debug('ODP event queue has been disabled.')
```

Comment:
```suggestion
            self.logger.debug('ODP is not integrated.')
```
Can we add this error to enums like other standard error messages?